pkg/cloud/externalconn/connectionpb/connection.proto

```diff
@@ -30,6 +30,7 @@ message ConnectionDetails {
   oneof details {
```

Comment:
@miretskiy a more meta question not directly related to this PR, but if we start having these top-level details for each provider we're going to have 10-12 proto messages with URI as their only field to begin with. I'm thinking we change this to to be less verbose and then switch on the provider:

```
message ConnectionDetails {
  ConnectionProvider provider = 1;
  string URI = 2;

  oneof details {
      // Optional sink specific details if a sink has more than a URI.
  }
```

wdyt?

Reply:
Perhaps just have a wrapper type (SimpleURI) that you can stick inside oneof?

Reply:
Opened https://github.com/cockroachdb/cockroach/pull/85581